lib/ecto/model/optimistic_locking.ex

```diff
@@ -0,0 +1,18 @@
+defmodule Ecto.Model.OptimisticLocking do
+  defmacro optimistic_locking(field) do
+    hook_name = String.to_atom("optimistic_locking_#{field}")
```

Comment:
Don't do this here otherwise we are assuming an atom is given at compilation time. Do quote `bind_quoted: [field: field]` and convert it inside the quote. Also, you can simply write `:"optimistic_locking_#{field}"`. :)